### ISSUE TITLE
Add temperature to softmax

### ISSUE THREAD

**Issue by csukuangfj** [k2-fsa/sherpa#459]
The steps to do that are given below.

## Streaming models
1. Add a member after the following line
https://github.com/k2-fsa/sherpa/blob/4254d4a302bc7bc2497900d7474dcc29bbc23b9f/sherpa/cpp_api/online-recognizer.h#L70

```cpp
// temperature for the softmax in the joiner
float temperature = 1.0;
```

2. Register a commandline argument for the added member by adding a line after
https://github.com/k2-fsa/sherpa/blob/4254d4a302bc7bc2497900d7474dcc29bbc23b9f/sherpa/cpp_api/online-recognizer.cc#L110-L115

3. Change
https://github.com/k2-fsa/sherpa/blob/4254d4a302bc7bc2497900d7474dcc29bbc23b9f/sherpa/cpp_api/online-recognizer.cc#L175

```cpp
os << "chunk_size=" << chunk_size << ", ";
os << "temperature=" << chunk_size << ")";
```

4. Fix the python binding by changing
https://github.com/k2-fsa/sherpa/blob/4254d4a302bc7bc2497900d7474dcc29bbc23b9f/sherpa/python/csrc/online-recognizer.cc#L28
to
```cpp
 const FastBeamSearchConfig &fast_beam_search_config = {}, float temperature = 1.0) 
```
Change
https://github.com/k2-fsa/sherpa/blob/4254d4a302bc7bc2497900d7474dcc29bbc23b9f/sherpa/python/csrc/online-recognizer.cc#L47
to
```cpp
 ans->chunk_size = chunk_size; 
 ans->temperature = temperature;
```

Change
https://github.com/k2-fsa/sherpa/blob/4254d4a302bc7bc2497900d7474dcc29bbc23b9f/sherpa/python/csrc/online-recognizer.cc#L60
to
```cpp
 py::arg("fast_beam_search_config") = FastBeamSearchConfig(),
 py::arg("temperature") = 1.0) 
```

Change
https://github.com/k2-fsa/sherpa/blob/4254d4a302bc7bc2497900d7474dcc29bbc23b9f/sherpa/python/csrc/online-recognizer.cc#L77
to
```cpp
 .def_readwrite("chunk_size", &PyClass::chunk_size) 
 .def_readwrite("temperature", &PyClass::temperature) 
```

5. Change
https://github.com/k2-fsa/sherpa/blob/4254d4a302bc7bc2497900d7474dcc29bbc23b9f/sherpa/csrc/online-transducer-modified-beam-search-decoder.h#L36
to
```cpp
  int32_t num_active_paths_;
  float temperature = 1.0;
```

Change
https://github.com/k2-fsa/sherpa/blob/4254d4a302bc7bc2497900d7474dcc29bbc23b9f/sherpa/csrc/online-transducer-modified-beam-search-decoder.h#L18-L19
to
```
      OnlineTransducerModel *model, int32_t num_active_paths, float temperature)
      : model_(model), num_active_paths_(num_active_paths), temperature_(temperature) {}
```

6. Change
https://github.com/k2-fsa/sherpa/blob/4254d4a302bc7bc2497900d7474dcc29bbc23b9f/sherpa/cpp_api/online-recognizer.cc#L299
to
```cpp
model_.get(), config.num_active_paths, config.temperature);
```

7. Change
https://github.com/k2-fsa/sherpa/blob/4254d4a302bc7bc2497900d7474dcc29bbc23b9f/sherpa/csrc/online-transducer-modified-beam-search-decoder.cc#L172
```
auto log_probs = (logits / temperature_).log_softmax(-1).cpu();
```

That's it!

# Non-streaming models

Please follow the above steps and change `online` to `offline` and modify the corresponding files.
